### PR TITLE
Automatically reconnect if server connection drops

### PR DESCRIPTION
**Pull request by ihull**
Also minor change to format JS Date in results as string (this will need to be improved but will require enhancement to node-q module)

### CODE REVIEW

**Comment by ihull** [comment]
Commit f36bd44:

- Improve user feedback for queries that don't return a result: show server response (often the query echoed back) and date/time
- For non-tabulated results, such as an array of numbers, show table as a single column of values (as per the original KDB Studio)

File: components/query-results.js [hunk 1]
```diff
@@ -1,4 +1,17 @@
 const { autoUpdater } = require("electron");
+const { dates } = require("node-q");
```
Comment (ColinEberhardt) [inline components/query-results.js:2]: (minor) unused import?

File: components/query-results.js [hunk 2]
```diff
@@ -50,23 +63,34 @@ const queryResults = {
...
-                    minWidth: "20px",
-                  };
-                });
-              } else {
-                for (let x in data) {
-                  outputHTML += x + " | " + data[x] + "<br />";
+                if (data[0] instanceof Object) {
+                  const colnames = Object.keys(data[0]);
+                  this.tableData = formatData(colnames, data);
+                  this.columns = colnames.map((c) => {
```
Comment (ColinEberhardt) [inline components/query-results.js:69]: Quick tip, you can simplify further:

~~~
map((c) => {
 return { foo: "bar" };
});
~~~

is equivalent to:

~~
map((c) => ({ foo: "bar" }));
~~~

**Comment by ColinEberhardt** [comment]
@ihull looks good to me - a couple of minor comments, see above.

File: components/query-results.js (hunk 1 above)
Comment (ihull) [reply components/query-results.js:2]: Ah yes

File: components/query-results.js (hunk 2 above)
Comment (ihull) [reply components/query-results.js:69]: Thanks!

**Pull request merged**